app/controllers/journeys/nics/Class2NICsController.scala

```diff
@@ -53,8 +54,10 @@ class Class2NICsController @Inject() (override val messagesApi: MessagesApi,
 
   def onSubmit(taxYear: TaxYear, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
     def handleError(formWithErrors: Form[_]): Result = BadRequest(view(formWithErrors, taxYear, request.userType, mode))
+    def handleSuccess(answer: Boolean): Future[Result] =
+      service.submitGatewayQuestionAndRedirect(page, nationalInsuranceContributions, request.userAnswers, answer, taxYear, mode)
```

Comment:
This is to ensure that any Class 4 data is being cleared. Using the defaultHandleForm method doesn't clear data for boolean pages